### PR TITLE
[FLINK-23088] Add flink-rpc-core module

### PR DESCRIPTION
**Pull request by zentol**
Based on #16265.

Adds a new flink-rpc-core module, acting as an interface between flink-runtime and the future flink-rpc-akka module. Essentially we're just moving the core rpc interface (like RpcService) to a new module, with some checkstyle fixes being applied on the way.

### CODE REVIEW

**Comment by XComp** [comment]
Can we close this PR considering that I went over the changes in #16312 already?

**Comment by zentol** [comment]
I wouldn't do that. For one it wasn't obvious to me _at all_ that you also looked at the flink-rpc-core module, and it is also nicer from an audit perspective if the PR opened for an issue is then also the one via which the changes are merged.

File: flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryConfiguration.java [hunk 1]
```diff
@@ -76,7 +73,8 @@ public long getQueryServiceMessageSizeLimit() {
      * @param configuration to generate the metric registry configuration from
```
Comment (XComp) [inline flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryConfiguration.java:73]: Missing JavaDoc for `maximumFrameSize`

**Review by XComp — APPROVED** [review]
Ok, I did another pass over the changes. I didn't find anything except for a missing JavaDoc tag. Thanks @zentol

File: flink-runtime/src/test/java/org/apache/flink/runtime/metrics/MetricRegistryTestUtils.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+package org.apache.flink.runtime.metrics;
+
+import org.apache.flink.configuration.Configuration;
+
+/** Test utilities for the metric registries. */
+public class MetricRegistryTestUtils {
+
+    private static volatile MetricRegistryConfiguration defaultConfiguration;
```
[the remark this answers is not in the document]
Comment (zentol) [reply flink-runtime/src/test/java/org/apache/flink/runtime/metrics/MetricRegistryTestUtils.java:25]: Minor optimizations it seems. in production this isn't used and we don't want to create this object for no reason.

**Pull request merged**